src/irr/asset/CImageLoaderOpenEXR.cpp

```diff
@@ -0,0 +1,454 @@
+							assignToReinterpretedValue(params.format, fetchedData, ptrStyleEndShiftToImageDataPixel + channelIndex, (fullFloatPixelMapArray[channelIndex])[xPos][yPos]);
+						else if(params.format == EF_R32G32B32A32_UINT || params.format == EF_R32G32B32_UINT)
+							assignToReinterpretedValue(params.format, fetchedData, ptrStyleEndShiftToImageDataPixel + channelIndex, (uint32_tPixelMapArray[channelIndex])[xPos][yPos]);
+					}
+
+			image->setBufferAndRegions(std::move(texelBuffer), regions);
+
+			return SAssetBundle{image};
+		}
+
+		bool CImageLoaderOpenEXR::isALoadableFileFormat(io::IReadFile* _file) const
+		{
+			/*
+
+			TODO!
+			because of old versions don't have a file layout with magic number as
+			first 4 bytes, I have to improve it
+			
+			const size_t begginingOfFile = _file->getPos();
+
+			unsigned char magicNumberBuffer[sizeof(SContext::magicNumber)];
+			_file->read(magicNumberBuffer, sizeof(SContext::magicNumber));
+			_file->seek(begginingOfFile);
+
+			auto deserializeToReadMagicValue = [&](unsigned char* buffer)
+			{
+				uint32_t value = 0ul;
+				value |= buffer[0] << 24 | buffer[1] << 16 | buffer[2] << 8 | buffer[3];
+				return value;
+			};
+
+			auto magicNumberToCompare = deserializeToReadMagicValue(magicNumberBuffer);
+			if (magicNumberToCompare == SContext::magicNumber)
+				return true;
+			else
+				return false; */
+			return true;
+		}
+
+		template<typename rgbaFormat>
+		void readRgba(InputFile& file, std::array<Array2D<rgbaFormat>, 4>& pixelRgbaMapArray, int& width, int& height, E_FORMAT& format, const bool& doesItSupportAlphaChannel)
+		{
+			Box2i dw = file.header().dataWindow();
+			width = dw.max.x - dw.min.x + 1;
+			height = dw.max.y - dw.min.y + 1;
+
+			const std::string rgbaSignatureAsText = "RGBA";
+			for (auto& pixelChannelBuffer : pixelRgbaMapArray)
+				pixelChannelBuffer.resizeErase(height, width);
+
+			FrameBuffer frameBuffer;
+			PixelType pixelType;
+
+			if (format == EF_R16G16B16A16_SFLOAT || format == EF_R16G16B16_SFLOAT)
+				pixelType = PixelType::HALF;
+			else if (format == EF_R32G32B32A32_SFLOAT || format == EF_R32G32B32_SFLOAT)
+				pixelType = PixelType::FLOAT;
+			else if (format == EF_R32G32B32A32_UINT || format == EF_R32G32B32_UINT)
+				pixelType = PixelType::UINT;
+
+			for (uint8_t rgbaChannelIndex = 0; rgbaChannelIndex < (doesItSupportAlphaChannel ? 4 : 3); ++rgbaChannelIndex)
+				frameBuffer.insert
+				(
+					reinterpret_cast<const char*>(rgbaSignatureAsText[rgbaChannelIndex]),                       // name
+					Slice(pixelType,                                                                            // type
+					(char*)(&(pixelRgbaMapArray[rgbaChannelIndex])[0][0] - dw.min.x - dw.min.y * width),        // base
+					sizeof((pixelRgbaMapArray[rgbaChannelIndex])[0][0]) * 1,                                    // xStride
+					sizeof((pixelRgbaMapArray[rgbaChannelIndex])[0][0]) * width,                                // yStride
```

Comment:
are you sure the strides are in pixels and not bytes?

Reply:
everything is done well, I read a few pages of their documentation